### PR TITLE
update metadata for 1.14.0

### PR DESCRIPTION
**Pull request by zacharyburnett**
waiting for the following before unit and regression testing can start:
- [x] `stpipe 0.5.2`
    - [x] https://github.com/spacetelescope/stpipe/pull/149
- [x] `stdatamodels 1.10.1`
    - [x] https://github.com/spacetelescope/stdatamodels/pull/268
    - [x] https://github.com/spacetelescope/stdatamodels/pull/270
    - [x] https://github.com/spacetelescope/stdatamodels/pull/288
- [x] ~`stcal 1.6.2`~ `stcal 1.7.0`
    - [x] https://github.com/spacetelescope/stcal/pull/248
    - [x] #8303
    - [x] https://github.com/spacetelescope/stcal/pull/251
- [x] #8304
- [x] #8365
- [x] successful completion of https://plwishmaster.stsci.edu:8081/blue/organizations/jenkins/RT%2FJWST/detail/JWST/2829/pipeline/
 
**Checklist for maintainers**
- [x] ~added entry in `CHANGES.rst` within the relevant release section~
- [x] ~updated or added relevant tests~
- [x] updated relevant documentation
- [x] added relevant milestone
- [x] added relevant label(s)
- [x] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [x] ~Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)~

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -35,9 +35,9 @@ dependencies = [
     "scikit-image>=0.19",
     "scipy>=1.6.0",
     "spherical-geometry>=1.2.22",
-    "stcal>=1.6.0,<1.7.0",
-    "stdatamodels>=1.10.0,<1.11.0",
-    "stpipe>=0.5.0,<0.6.0",
+    "stcal>=1.6.1,<1.7.0",
```
Comment (zacharyburnett) [inline pyproject.toml:38]:
```suggestion
    "stcal>=1.6.2,<1.7.0",
```

File: pyproject.toml [hunk 2]
```diff
@@ -35,9 +35,9 @@ dependencies = [
     "scikit-image>=0.19",
     "scipy>=1.6.0",
     "spherical-geometry>=1.2.22",
-    "stcal>=1.6.0,<1.7.0",
-    "stdatamodels>=1.10.0,<1.11.0",
-    "stpipe>=0.5.0,<0.6.0",
+    "stcal>=1.6.2,<1.7.0",
```
Comment (hbushouse) [inline pyproject.toml:38]: This makes #8329 obsolete and could be closed.

**Comment by hbushouse** [comment]
Need to add #8303 and potentially #8365 to your check list above, as items we need/want in the first release.

**Comment by zacharyburnett** [comment]
one failed test:
https://plwishmaster.stsci.edu:8081/blue/organizations/jenkins/RT%2FJWST/detail/JWST/2829/pipeline/202#step-203-log-201
```
E           FileNotFoundError: [Errno 2] No such file or directory: '/data1/jenkins/workspace/RT/JWST/clone/test_outputs/popen-gw3/test_image3_closedfile0/jw00617-o082_t001_nircam_clear-f090w-sub320_i2d.fits'
```

File: CITATION.cff [hunk 3]
```diff
@@ -75,7 +75,7 @@ authors:
   given-names: "Maria"
   orcid: "https://orcid.org/0000-0003-2314-3453"
 title: "JWST Calibration Pipeline"
-version: 1.13.4
+version: 1.14.0
 doi: 10.5281/zenodo.7038885
-date-released: 2024-01-25
+date-released: 2024-03-22
```
Comment (zacharyburnett) [inline CITATION.cff:80]:
```suggestion
date-released: 2024-03-25
```

File: CHANGES.rst [hunk 4]
```diff
@@ -1,4 +1,4 @@
-1.13.5 (unreleased)
+1.14.0 (2024-03-22)
```
Comment (zacharyburnett) [inline CHANGES.rst:1]:
```suggestion
1.14.0 (2024-03-25)
```

File: README.md [hunk 5]
```diff
@@ -216,6 +216,7 @@ the specified context and less than the context for the next release.
 
 | jwst tag            | DMS build | SDP_VER  | CRDS_CONTEXT | Released   | Ops Install | Notes                                         |
 |---------------------|-----------|----------|--------------|------------|-------------|-----------------------------------------------|
+| 1.14.0              | B10.2rc1  |          | 1215         | 2024-03-22 |             | First release candidate for B10.2             |
```
Comment (zacharyburnett) [inline README.md:219]:
```suggestion
| 1.14.0              | B10.2rc1  |          | 1215         | 2024-03-25 |             | First release candidate for B10.2             |
```

**Comment by hbushouse** [comment]
> one failed test: https://plwishmaster.stsci.edu:8081/blue/organizations/jenkins/RT%2FJWST/detail/JWST/2829/pipeline/202#step-203-log-201
> 
> ```
> E           FileNotFoundError: [Errno 2] No such file or directory: '/data1/jenkins/workspace/RT/JWST/clone/test_outputs/popen-gw3/test_image3_closedfile0/jw00617-o082_t001_nircam_clear-f090w-sub320_i2d.fits'
> ```

Not an issue. That's one of the 3 or 4 tests that frequently fail for no obvious reason.

**Review by hbushouse — APPROVED** [review]

**Comment by hbushouse** [comment]
Didn't we want to include #7862 yet, before tagging master? Or is this just meta data update and actual 1.14.0 tag will come later?

**Comment by zacharyburnett** [comment]
> Didn't we want to include #7862 yet, before tagging master? Or is this just meta data update and actual 1.14.0 tag will come later?

I made a `1.14.0` tag but deleted it (there was no PyPI release so no harm done). I'll include #7862 when it's merged

**Pull request merged**